http/README.md

```diff
@@ -0,0 +1,113 @@
+
+*HTTP transport* is the thing that takes your high level request/response defined in terms of HTTP semantics and encodes it and sends it over the wire.
+
+When this document says *HTTP* it is generally referring to *HTTP semantics*.
+
+## Interoperability with existing HTTP systems
+
+A goal of this spec is to allow libp2p to be able to interoperate with existing HTTP servers and clients. Care is taken in this document to not introduce anything that would break interoperability with existing systems.
+
+## HTTP Transport
+
+Nodes MUST use HTTPS (i.e. they MUST NOT use plaintext HTTP). It is RECOMMENDED to use HTTP/2 and HTTP/3.
+
+Nodes signal support for their HTTP transport using the `/http` component in their multiaddr. e.g. `/dns4/example.com/tls/http` . See the [HTTP multiaddr component spec](https://github.com/libp2p/specs/pull/550) for more details.
+
+## Namespace
+
+libp2p does not squat the global namespace. libp2p application protocols can be discovered by the [well-known resource](https://www.rfc-editor.org/rfc/rfc8615) `.well-known/libp2p`. This allows server operators to dynamically change the URLs of the application protocols offered, and not hard-code any assumptions how a certain resource is meant to be interpreted.
+
+```json
+
+{
+    "services": {
+        "/kad/1.0.0": "/kademlia/",
+        "/ipfs-http/1.0.0": "/",
+    }
+}
+```
+
+The resource contains a mapping of application protocols to their respective URL. For example, this configuration file would tell a client
+
+1. That the Kademlia protocol is available at `/kademlia` and
+2. The [IPFS Path Gateway API](https://specs.ipfs.tech/http-gateways/path-gateway/) is mounted at `/`.
+
+It is valid to expose a service at `/`. It is RECOMMENDED that the server resolve more specific URLs before less specific ones. e.g. a path of `/kademlia/foo` should be routed to the Kademlia protocol rather than the IPFS HTTP API.
+
+## Peer ID Authentication
+
+When using the HTTP Transport, peer id authentication is optional. You only pay for it if you need it. This benefits use cases that don’t need peer authentication (e.g. fetching content addressed data) or authenticate some other way (not tied to libp2p peer ids).
+
+Peer ID authentication in the HTTP Transport follows a similar to pattern to how
+libp2p adds Peer ID authentication in WebTransport and WebRTC. We run the
+standard libp2p Noise handshake, but using `IX` for client and server
+authentication or `NX` for just server authentication.
+
+Note: This is just one form of Peer ID authentication. Other forms may be added
+in the future (with a different `www-authenticate` value) or be added to the
+application protocols themselves.
+
+### Authentication flow
+
+1. The client initiates a request that it knows must be authenticated OR the client responds to a `401` with the header `www-authenticate: libp2p-noise` (The server MAY also include `libp2p-token` as an authentication scheme).
```

Comment:
nit: other headers follow the canonical form, with each work capitalized:
```suggestion
1. The client initiates a request that it knows must be authenticated OR the client responds to a `401` with the header `WWW-Authenticate: libp2p-noise` (The server MAY also include `libp2p-token` as an authentication scheme).
```

While at it.. I wonder if we should also capitalize `Libp2p-Noise` and `Libp2p-Token` while at it, since preexisting values are `Basic` and `Digest`. Most HTTP software is having normalization code in place for headers anyway, like golang's https://pkg.go.dev/net/http#CanonicalHeaderKey